File: JS/src/move_zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
function moveZeroes (nums) {
  let slow = 0
  let fast = 0

  while (fast < nums.length) {
    if (nums[fast] !== 0) {
      const temp = nums[fast]
      nums[fast] = nums[slow]
      nums[slow] = temp
      slow++
    }
    fast++
  }
}

export { moveZeroes }

File: C++/src/move_zeroes.cc
#include "move_zeroes.hpp"

void Solution::moveZeroes(vector<int> &nums) {
  size_t slow = 0;
  size_t fast = 0;

  while (fast < nums.size()) {
    if (nums[fast] != 0) {
      swap(nums[fast++], nums[slow++]);
    } else {
      fast++;
    }
  }
}